data/rulesets.ts

```diff
@@ -1085,21 +1086,27 @@ export const Rulesets: {[k: string]: FormatData} = {
 							}
 							const forme = dex.species.get(formeName);
 							if (forme.changesFrom === originalForme.name && !forme.battleOnly) {
-								types = types.concat(forme.types);
+								speciesTypes.push(...forme.types);
 							}
 						}
 					}
 				} else {
-					types = species.types;
+					speciesTypes.push(...species.types);
 				}
 
 				let prevo = species.prevo;
 				while (prevo) {
 					const prevoSpecies = dex.species.get(prevo);
-					types = types.concat(prevoSpecies.types);
+					speciesTypes.push(...prevoSpecies.types);
 					prevo = prevoSpecies.prevo;
 				}
-				if (types.includes(move.type)) return null;
+				// Check for type changes from past generations
+				for (let i = dex.gen - 1; i >= species.gen && i >= move.gen; i--) {
+					const genDex = dex.forGen(i);
+					speciesTypes.push(...genDex.species.get(species.name).types);
+					moveTypes.push(genDex.moves.get(move.name).type);
+				}
+				if (moveTypes.some(m => speciesTypes.includes(m))) return null;
```

Comment:
Should I reimplement the forme and prevo logic here? It's not required for the current mons this change affects, but maybe for future proofing?

Reply:
nevermind just figured out a better way to implement the generation loop